### ISSUE TITLE
Should we split coder_spec into multiple files?

### ISSUE THREAD

**Issue by hugomg** [pallene-lang/pallene#80]
Currently, any test that involves executing some Pallene code lives inside coder_spec. Should we split this into smaller files? For example, a separate file for builtin functions?

**Comment by hugomg** [comment]
Currently the builtin functions like table.insert  are implemented inside the coder, and they are effectively being inlined into the generated code. I think perhaps it would make more sense to split them into a separate test file if they were implemented as regular functions in a standard library.

That said, the coder_spec file is getting pretty long. Would splitting it into smaller things be a good idea or are describe() blocks good enough already?

**Comment by gligneul** [comment]
Not only coder spec is long, but the coder.lua itself is enormous.
I get lost in the file every time I have to edit it.
I think we should split both of them in smaller files.

**Comment by hugomg** [comment]
I can imagine it being a good idea to split some of the subroutines from coder.lua in a separate file but I wouldn't want to split it into separate sequential passes or something like that.

Anyway, the ongoing IR refactor will lead to heavy rewrites in the coder and hopefully it will result in it being a little bit less confusing :)

**Comment by hugomg** [comment]
It would be nice if as part of this change we modified all the spec files to always invoke the compiler "all the way to the end", which would let us get rid of or drastically simplify the `driver.compile_internal` function.

I think the only obstacle to this are the "non error" test cases in parser_spec. The interesting tests could be rewritten into tests that run a program (like the ones in coder_spec). The more trivial parser tests I think we can just throw away.

**Comment by hugomg** [comment]
One potential benefit of splitting the coder_spec into multiple files is that it would speed up the test suite. Since we started parallelizing it, the current bottleneck is how long it takes to run the coder_spec. If it were split into multiple files, it would be possible to run those in parallel as well.

This suggests that splitting the coder_spec could be helpful even if we don't split the coder.lua itself.
